### PR TITLE
[BUGFIX beta] Partially revert #12229 and add a test for #13432

### PR DESCRIPTION
**Pull request by Serabe**
This PR partially reverts #12229 given that it made the inline form of
components extending from `LinkTo` impossible.

Beware that matching the exact behaviour is not there yet (as it wasn't
before #12229) given that `{{{link-to title route}}}` would unescape `title`
while `{{{my-link-to title route}}}` would not. This behaviour was not working
before and therefore is not working after this revert.

This PR does not address whether the inline form for `link-to` should be
deprecated or not.

### CODE REVIEW

**Comment by Serabe** [comment]
OTOH, I think the behaviour for triple curlies in `link-to` should be deprecated given:

* It is a hack not easily replicable in the component itself.
* Makes `link-to` behave differently from other components.
* That behaviour cannot be ported easily to all components extending from `link-to`.
* It can be easily transformed from `{{{link-to title route}}}` to `{{#link-to route}} {{{title}}} {{/link-to}}`.

**Comment by rwjblue** [comment]
Ultimately, I'd like to deprecate this inline form for custom `Ember.LinkComponent` classes (and possibly even for all inline `link-to`), so leaving the AST transform for `{{link-to 'thing' 'thing'}}` but bringing back support in `packages/ember-routing-views/lib/components/link-to.js` allows us to support the old behavior and still deprecate custom `LinkComponent` inline form.

*tldr;* Leave the AST transform added in https://github.com/emberjs/ember.js/pull/12229 in, add the test you have added here, and revert the changes https://github.com/emberjs/ember.js/pull/12229 made to `packages/ember-routing-views/lib/components/link-to.js`.

**Comment by Serabe** [comment]
Updated! Thank you for the quick response!

**Comment by rwjblue** [comment]
Awesome, this LGTM.  We are having an issue with Sauce Labs ATM (they aren't recognizing one of the Firefox builds).  I'll try to dig into that later today.

**Comment by rwjblue** [comment]
https://github.com/emberjs/ember.js/pull/13439 should have fixed the build, thanks for rebasing so quickly

**Comment by rwjblue** [comment]
Thanks @Serabe!
